src/NServiceBus.Core/Routing/RoutingMappingSettings.cs

```diff
@@ -32,13 +31,13 @@ internal RoutingMappingSettings(SettingsHolder settings)
         public Publishers Publishers => GetOrCreate<Publishers>();
 
         /// <summary>
-        /// Sets a distribution strategy for a given subset of message types.
+        /// Sets a distribution strategy for a given endpoint.
         /// </summary>
+        /// <param name="endpointName">The name of the logical endpoint the given strategy should apply to.</param>
         /// <param name="distributionStrategy">The instance of a distribution strategy.</param>
-        /// <param name="typeMatchingRule">A predicate for determining the set of types.</param>
-        public void SetMessageDistributionStrategy(DistributionStrategy distributionStrategy, Func<Type, bool> typeMatchingRule)
+        public void SetMessageDistributionStrategy(string endpointName, DistributionStrategy distributionStrategy)
```

Comment:
I thought about a way to change the strategy too but I think this may be a bit overkill for now (shouldn't be too difficult to implement once users need it). we could provide a `SetMessageDistributionStrategy` which takes a `Func<string, ...>` to resolve the strategy but that would mean we have to evaluate these rules on each routing operation again :/